Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI.xcodeproj/project.pbxproj

```diff
@@ -81,7 +85,6 @@
 		D544FB242772102F0015925C /* MapView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = MapView.swift; sourceTree = "<group>"; };
 		D544FB262772103B0015925C /* Info.plist */ = {isa = PBXFileReference; lastKnownFileType = text.plist; path = Info.plist; sourceTree = "<group>"; };
 		D544FB46277332760015925C /* MapViewModel.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = MapViewModel.swift; sourceTree = "<group>"; };
-		D544FB482773338B0015925C /* Secrets.xcconfig */ = {isa = PBXFileReference; lastKnownFileType = text.xcconfig; name = Secrets.xcconfig; path = ../../Secrets.xcconfig; sourceTree = "<group>"; };
```

Comment:
My first observation is that the app is now crashing when I try to create a publisher due to the lack of a Mapbox access token. I imagine it's due to this removal. Why did you remove it?

```
MapboxDirections/Credentials.swift:56: Precondition failed: A Mapbox access token is required. Go to <https://account.mapbox.com/access-tokens/>. In Info.plist, set the MBXAccessToken key to your access token, or use the Directions(accessToken:host:) initializer.
2022-10-25 08:40:29.236315-0300 PublisherExampleSwiftUI[43062:1189627] MapboxDirections/Credentials.swift:56: Precondition failed: A Mapbox access token is required. Go to <https://account.mapbox.com/access-tokens/>. In Info.plist, set the MBXAccessToken key to your access token, or use the Directions(accessToken:host:) initializer.
(lldb) bt
* thread #1, queue = 'com.apple.main-thread', stop reason = Precondition failed: A Mapbox access token is required. Go to <https://account.mapbox.com/access-tokens/>. In Info.plist, set the MBXAccessToken key to your access token, or use the Directions(accessToken:host:) initializer.
    frame #0: 0x000000018bf1d8b0 libswiftCore.dylib`_swift_runtime_on_report
    frame #1: 0x000000018bfaa550 libswiftCore.dylib`_swift_stdlib_reportFatalErrorInFile + 204
    frame #2: 0x000000018bc596d4 libswiftCore.dylib`closure #1 (Swift.UnsafeBufferPointer<Swift.UInt8>) -> () in closure #1 (Swift.UnsafeBufferPointer<Swift.UInt8>) -> () in Swift._assertionFailure(_: Swift.StaticString, _: Swift.String, file: Swift.StaticString, line: Swift.UInt, flags: Swift.UInt32) -> Swift.Never + 192
    frame #3: 0x000000018bc58598 libswiftCore.dylib`Swift._assertionFailure(_: Swift.StaticString, _: Swift.String, file: Swift.StaticString, line: Swift.UInt, flags: Swift.UInt32) -> Swift.Never + 300
  * frame #4: 0x00000001028872d4 PublisherExampleSwiftUI`Credentials.init(token="", host=nil) at Credentials.swift:56:9
    frame #5: 0x000000010243ef74 PublisherExampleSwiftUI`MapboxConfiguration.getCredentials(self=(mapboxKey = "")) at MapboxConfiguration+Extensions.swift:6:16
    frame #6: 0x0000000102443304 PublisherExampleSwiftUI`DefaultLocationService.init(mapboxConfiguration=(mapboxKey = ""), historyLocation=nil, logHandler=0x0000600001be0a00, vehicleProfile=car, self=0x0000600001c8e3a0) at DefaultLocationService.swift:18:70
    frame #7: 0x0000000102443148 PublisherExampleSwiftUI`DefaultLocationService.__allocating_init(mapboxConfiguration:historyLocation:logHandler:vehicleProfile:) at DefaultLocationService.swift:0
    frame #8: 0x000000010242edc0 PublisherExampleSwiftUI`DefaultPublisherBuilder.start(self=0x0000600000dbca80) at DefaultPublisherBuilder.swift:72:60
    frame #9: 0x00000001024314d4 PublisherExampleSwiftUI`protocol witness for PublisherBuilder.start() in conformance DefaultPublisherBuilder at <compiler-generated>:0
    frame #10: 0x000000010240056c PublisherExampleSwiftUI`CreatePublisherViewModel.createPublisher(self=0x00006000016b7380) at CreatePublisherViewModel.swift:85:14
    frame #11: 0x00000001023f6f44 PublisherExampleSwiftUI`closure #1 in closure #7 in closure #3 in closure #1 in CreatePublisherView.body.getter(self=PublisherExampleSwiftUI.CreatePublisherView @ 0x0000600000a989a0) at CreatePublisherView.swift:112:47
    frame #12: 0x0000000111a2ee5c SwiftUI`___lldb_unnamed_symbol157184 + 20
    frame #13: 0x00000001119df600 SwiftUI`___lldb_unnamed_symbol154089 + 20
    frame #14: 0x000000011193faa4 SwiftUI`___lldb_unnamed_symbol148792 + 56
    frame #15: 0x0000000111add34c SwiftUI`___lldb_unnamed_symbol163418 + 1096
    frame #16: 0x0000000111adf0a0 SwiftUI`___lldb_unnamed_symbol163430 + 72
    frame #17: 0x0000000107e70cc0 UIKitCore`-[UICollectionView _selectItemAtIndexPath:animated:scrollPosition:notifyDelegate:deselectPrevious:performCustomSelectionAction:] + 1056
    frame #18: 0x0000000107e9be98 UIKitCore`-[UICollectionView touchesEnded:withEvent:] + 476
    frame #19: 0x00000001086995f0 UIKitCore`forwardTouchMethod + 264
    frame #20: 0x00000001086995f0 UIKitCore`forwardTouchMethod + 264
    frame #21: 0x00000001086995f0 UIKitCore`forwardTouchMethod + 264
    frame #22: 0x00000001086995f0 UIKitCore`forwardTouchMethod + 264
    frame #23: 0x000000010821b29c UIKitCore`_UIGestureEnvironmentUpdate + 5876
    frame #24: 0x00000001082198c8 UIKitCore`-[UIGestureEnvironment _deliverEvent:toGestureRecognizers:usingBlock:] + 276
    frame #25: 0x0000000108219644 UIKitCore`-[UIGestureEnvironment _updateForEvent:window:] + 156
    frame #26: 0x00000001086a8594 UIKitCore`-[UIWindow sendEvent:] + 3168
    frame #27: 0x0000000108686704 UIKitCore`-[UIApplication sendEvent:] + 692
    frame #28: 0x0000000108702b04 UIKitCore`__dispatchPreprocessedEventFromEventQueue + 6828
    frame #29: 0x0000000108704768 UIKitCore`__processEventQueue + 5612
    frame #30: 0x00000001086fd70c UIKitCore`__eventFetcherSourceCallback + 220
    frame #31: 0x0000000180371070 CoreFoundation`__CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 24
    frame #32: 0x0000000180370fb8 CoreFoundation`__CFRunLoopDoSource0 + 172
    frame #33: 0x0000000180370728 CoreFoundation`__CFRunLoopDoSources0 + 232
    frame #34: 0x000000018036ae68 CoreFoundation`__CFRunLoopRun + 756
    frame #35: 0x000000018036a75c CoreFoundation`CFRunLoopRunSpecific + 584
    frame #36: 0x0000000188f60c98 GraphicsServices`GSEventRunModal + 160
    frame #37: 0x000000010866cb74 UIKitCore`-[UIApplication _run] + 868
    frame #38: 0x0000000108670b1c UIKitCore`UIApplicationMain + 124
    frame #39: 0x0000000111f26130 SwiftUI`___lldb_unnamed_symbol197918 + 164
    frame #40: 0x0000000111f25fd8 SwiftUI`___lldb_unnamed_symbol197916 + 140
    frame #41: 0x0000000111709894 SwiftUI`static SwiftUI.App.main() -> () + 80
    frame #42: 0x00000001023aaf34 PublisherExampleSwiftUI`static PublisherExampleSwiftUIApp.$main(self=PublisherExampleSwiftUI.PublisherExampleSwiftUIApp) at PublisherExampleSwiftUIApp.swift:3:1
    frame #43: 0x00000001023aafdc PublisherExampleSwiftUI`main at PublisherExampleSwiftUIApp.swift:0
    frame #44: 0x00000001043b5fa0 dyld_sim`start_sim + 20
    frame #45: 0x00000001044a1e50 dyld`start + 2544
(lldb) 
```

Reply:
Hmm, I think that CI was failing because of it. Let me try to re-add it.

Reply:
Build is now failing for me locally as of 392b2461c261e2c08611c9291f5771dfc2f5f970:

```
/Users/lawrence/code/work/ably/ably-asset-tracking-swift/Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI/Secrets.xcconfig error build: /Users/lawrence/code/work/ably/ably-asset-tracking-swift/Examples/PublisherExampleSwiftUI/PublisherExampleSwiftUI/Secrets.xcconfig: No such file or directory
```

Reply:
I'd suggest just restoring the project file from `main` and adding your two files.

Reply:
Re-added in [392b246](https://github.com/ably/ably-asset-tracking-swift/pull/438/commits/392b2461c261e2c08611c9291f5771dfc2f5f970). Hopefully CI doesn't explode this time.

Reply:
But now in the diff of the PR there's still a change that I need to try and understand - it'll be easier for me to review if there's no change related to `Secrets.xcconfig` at all, which is why I suggested restoring the original file.

Reply:
Hmm, so I used a version from main branch in [a4a4d97](https://github.com/ably/ably-asset-tracking-swift/pull/438/commits/a4a4d97ec5858aa3905a0ea5a08a294d6814bbff), and manually re-added my new files. But now the Secrets.swift file is not detected by Xcode on my machine. I have to delete this red file, and re-add it manually. Then it works, but a diff is created in pbxproj file and apparently it stops working for you.

Maybe the local paths to the file are different on our machines, and Xcode has some troubles with that, (since the Secrets file itself is not pushed to the repo?)... I won't have time to investigate it before my holidays, so I'll leave it as is for now.
<img width="312" alt="Zrzut ekranu 2022-10-25 o 15 23 20" src="https://user-images.githubusercontent.com/15122679/197785508-36c79166-45d0-4865-a60e-393141af143e.png">

Reply:
No worries, I'll get this one tidied up and merged (it's working for me and CI), and when you're back you can take a look into what's going on on your machine, OK?

Reply:
Sounds good, thanks!